charts/backingservices/charts/srs/templates/_helpers.tpl

```diff
@@ -0,0 +1,150 @@
+{{/*
+App name
+*/}}
+{{- define "srs.name" -}}
+{{- default .Root.Chart.Name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified deployment name. This is used to define the deployment resource name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "srs.fullname" -}}
+{{- if .Values.deploymentName }}
+{{- .Values.deploymentName | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.deploymentName }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "srs.chart" -}}
+{{- $name := "srs" }}
+{{- $version := "0.1.0" }}
+{{- printf "%s-%s" $name $version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Common labels for all k8s resources
+*/}}
+{{- define "srs.labels" }}
+helm.sh/chart: {{ include "srs.chart" . }}
+{{ include "srs.selectorLabels" . }}
+{{- if .Root.Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Root.Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Root.Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels using app and release name
+*/}}
+{{- define "srs.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "srs.name" . }}
+app.kubernetes.io/instance: {{ .Root.Release.Name }}
+{{- end }}
+
+{{/*
+srs-service meta-data labels
+*/}}
+{{- define "srs.srs-service.labels" -}}
+{{ $data := dict "Root" $ "Name" "srs-service" }}
+{{- include "srs.labels" $data }}
+{{- end -}}
+
+{{/*
+srs-service match labels
+*/}}
+{{- define "srs.srs-service.match-labels" }}
+app.kubernetes.io/name: srs-service
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{ if .Values.srsStorage.requireInternetAccess -}}
+networking/allow-internet-egress: "true"
```

Comment:
Is this annotation valid with both internal elasticsearch cluster provisioning and with external Elasticsearch cluster URL is provided?

Reply:
this is valid with external es only and client can choose to enable this or not only based on whether srs needs to egress over internet to connect to the external es.

Reply:
currently looks like there is no check for "provisionInternalESCluster" is true or false. This annotation is directly added if 
.Values.srsStorage.requireInternetAccess is present irrespective of internal or external es cluster. Does it make sense to update the if condition to check provisionInternalESCluster and .Values.srsStorage.requireInternetAccess ?

Reply:
done